lib/RAST_SDK/RAST_SDKImpl.pm

```diff
@@ -1469,6 +1470,13 @@ sub annotate_genomes
 	    call_features_prophage_phispy => 1,
 	    retain_old_anno_for_hypotheticals => 1
 	});
+	
+	my $empty_input_msg = ("Required inputs: genomes in an array with at least one genome;\n".
+	                       "and/or a list of genome names separated by ';', '\n' or '|' (without quotes).\n";
+	Bio::KBase::Exceptions::ArgumentValidationError->throw(
+            error        => $empty_input_msg,
+            method_name  => 'annotate_genomes'
+        ) unless (is_arrayref( $genomes ) && @$genomes) && defined($params->{genome_text});
```

Comment:
No need for all the extra brackets:

```
... unless is_arrayref $genomes && @$genomes && defined $params->{ genome_text };
```

If `0` and `undef` are not valid values for `$params->{ genome_text }`, you can change the last part to 
```
... && $params->{ genome_text };
```

Reply:
To me this looks like code for requiring both a genome in the array AND genomes in the list. Am I wrong here?

Reply:
Where is `$genomes` initialised?

Reply:
A few lines later

Reply:
sorry Miriam, I meant the question for QZ. :)

Reply:
Well, it was a good question.

Reply:
I think you need to add something form genome_text being defined but empty.

Reply:
I added the test portion to my test file in test/testMetagutil.pl and ran the tests.  I only test the error message throwing part.